mlir/test/Dialect/Affine/parallelize.mlir

```diff
@@ -0,0 +1,50 @@
+// RUN: mlir-opt %s -affine-parallelize | FileCheck %s
+
+// CHECK-DAG: [[MAP0:#map[0-9]+]] = affine_map<(d0, d1) -> (d0, d1)>
+// CHECK-DAG: [[MAP1:#map[0-9]+]] = affine_map<() -> (0)>
+// CHECK-DAG: [[MAP2:#map[0-9]+]] = affine_map<()[s0] -> (s0)>
+
+// CHECK-LABEL: func @loop_nest_3d_outer_two_parallel(
+func @loop_nest_3d_outer_two_parallel(%N : index) {
+// CHECK-SAME:  [[VAL_0:%.*]]: index) {
+  %0 = alloc() : memref<1024 x 1024 x vector<64xf32>>
+  %1 = alloc() : memref<1024 x 1024 x vector<64xf32>>
+  %2 = alloc() : memref<1024 x 1024 x vector<64xf32>>
+  affine.for %i = 0 to %N {
+    affine.for %j = 0 to %N {
+      affine.for %k = 0 to %N {
+        %5 = affine.load %0[%i, %k] : memref<1024x1024xvector<64xf32>>
+        %6 = affine.load %1[%k, %j] : memref<1024x1024xvector<64xf32>>
+        %7 = affine.load %2[%i, %j] : memref<1024x1024xvector<64xf32>>
+        %8 = mulf %5, %6 : vector<64xf32>
+        %9 = addf %7, %8 : vector<64xf32>
+        affine.store %9, %2[%i, %j] : memref<1024x1024xvector<64xf32>>
+      }
+    }
+  }
+  return
+}
+
+// CHECK:  %0 = alloc() : memref<1024x1024xvector<64xf32>>
+// CHECK:  %1 = alloc() : memref<1024x1024xvector<64xf32>>
+// CHECK:  %2 = alloc() : memref<1024x1024xvector<64xf32>>
+// CHECK:  affine.parallel (%[[arg1:.*]]) = (0) to (symbol(%arg0)) {
+// CHECK:    affine.parallel (%[[arg2:.*]]) = (0) to (symbol(%arg0)) {
+// CHECK:      affine.for %[[arg3:.*]] = 0 to %arg0 {
+
+// CHECK:  %3 = affine.load %0[%[[arg1:.*]], %[[arg3:.*]]] : memref<1024x1024xvector<64xf32>>
+// CHECK:  %4 = affine.load %1[%[[arg3:.*]], %[[arg2:.*]]] : memref<1024x1024xvector<64xf32>>
+// CHECK:  %5 = affine.load %2[%[[arg1:.*]], %[[arg2:.*]]] : memref<1024x1024xvector<64xf32>>
+// CHECK:  %6 = mulf %3, %4 : vector<64xf32>
```

Comment:
Avoid using numbers %3, %4 - because when the test case changes, it'd be hard to update everything. Instead use %{{.*}}. If capturing the value, use `%[[lhs:.*]] = affine.load` and then `%[[lhs]]` at all use sites.